src/main/rascal/refactor/Rename.rsc

```diff
@@ -184,38 +190,38 @@ RenameResult rename(
       , void(value at, str s) { registerMessage(error(at, s)); }
     );
 
-    printDebug("Renaming <cursor[0].src> to \'<newName>\'");
+    jobStart(config.jobLabel, totalWork = 2 * WORKSPACE_WORK);
 
-    printDebug("+ Finding definitions for cursor at <cursor[0].src>");
+    jobStep(config.jobLabel, "Resolving definitions of <cursor[0].src>", work = WORKSPACE_WORK);
     defs = getCursorDefinitions(cursor, parseLocCached, getTModelCached, r);
 
     if (defs == {}) r.error(cursor[0].src, "No definitions found");
-    if (errorReported()) return <sortDocEdits(docEdits), getMessages()>;
+    if (errorReported()) {
+        jobEnd(config.jobLabel, success=false);
+        return <sortDocEdits(docEdits), getMessages()>;
+    }
 
-    printDebug("+ Finding occurrences of cursor");
+    jobStep(config.jobLabel, "Looking for files with occurrences of name under cursor", work = WORKSPACE_WORK);
     <maybeDefFiles, maybeUseFiles> = findOccurrenceFiles(defs, cursor, parseLocCached, r);
 
-    if (maybeDefFiles != {}) {
-        printDebug("+ Finding additional definitions");
-        set[Define] additionalDefs = {};
-        for (loc f <- maybeDefFiles) {
-            printDebug("  - ... in <f>");
-            tr = parseLocCached(f);
-            tm = getTModelCached(tr);
-            fileAdditionalDefs = findAdditionalDefinitions(defs, tr, tm, r);
-            printDebug("    (found <size(fileAdditionalDefs)>)");
-            additionalDefs += fileAdditionalDefs;
-        }
-        defs += additionalDefs;
+    jobTodo(config.jobLabel, work = (size(maybeDefFiles) + size(maybeUseFiles)) * FILE_WORK);
+
+    set[Define] additionalDefs = {};
+    for (loc f <- maybeDefFiles) {
+        jobStep(config.jobLabel, "Looking for additional definitions in <f>", work = FILE_WORK);
+        tr = parseLocCached(f);
+        tm = getTModelCached(tr);
+        fileAdditionalDefs = findAdditionalDefinitions(defs, tr, tm, r);
+        additionalDefs += fileAdditionalDefs;
     }
+    defs += additionalDefs;
```

Comment:
Removed the `if`-condition around this since, for an empty list of `maybeDefFiles`, this should be instant anyway.